platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py

```diff
@@ -358,7 +358,33 @@ def get_change_event(self, timeout=0):
             status = self.sfp_event.check_sfp_status(port_dict, error_dict, timeout)
 
         if status:
-            self.reinit_sfps(port_dict)
+            if self.RJ45_port_list:
+                # Translate any plugin/plugout event into error dict
+                unknown_port = set()
+                unplug_port = set()
+                for index, event in port_dict.items():
+                    if index in self.RJ45_port_list:
+                        # Remove it from port event
+                        # check if it's unknown event
+                        if event == '0': # Remove event
+                            unplug_port.add(index)
+                        elif event != '1':
```

Comment:
will the "unkown" status also reported from PMPE or need to get it by calling "sx_api_port_status"?

Reply:
It should be. Otherwise there is inconvenience between PMPE and PMAOS, which IMO is a bug. Anyway, let’s double check.
The only exception AFAIK is the initialization flow. I don’t know whether the status change occurring before registered to the PMPE channel will be reported after registered. If not, we need to fetch the statuses for all the ports and report events for the ports that are unknown or not present when get_change_event is called for the first time.

Reply:
let's perform some tests for this after the holiday and I am merging your change for now.

Reply:
Currently, there is a bug in sdk. No matter which port is unplugged, it always reports port 2. Need to check it as well